README.md
[<img src="https://raw.githubusercontent.com/Majnu007/IMGHide-GUI/main/assets/header2.png">](https://github.com/Majnu007/IMGHide-GUI)

[<img src="https://img.shields.io/badge/LICENSE-Custom-blue?style=for-the-badge">](LICENSE) [<img src="https://img.shields.io/badge/Latest%20Version-1.3-orange?style=for-the-badge">](imghide-gui.py) [<img src="https://img.shields.io/github/repo-size/LinuxGuyy/IMGHide-GUI?label=Repository%20Size&style=for-the-badge">](https://github.com/Majnu007/IMGHide-GUI)

[IMGHide-GUI.py](imghide-gui.py) A FrontEnd For [IMGHide.py](https://github.com/teja156/imghide/blob/main/imghide.py) Which You Can Use To Hide secret texts/messages inside an image. You can optionally encrypt your texts with a password using AES-256 before encoding into the image.

---

## Screenshots Of The Software
Note Screenshots Are Old, they can differ from the Current version

[<img src="https://i.ibb.co/8ckgBwc/imghide-gui-1.png">](https://github.com/Majnu007/IMGHide-GUI)
[<img src="https://i.ibb.co/PC4mmH8/imghide-gui-2.png">](https://github.com/Majnu007/IMGHide-GUI)
[<img src="https://i.ibb.co/Bj5Hb7c/imghide-gui-3.png">](https://github.com/Majnu007/IMGHide-GUI)

---

## Installation
You can install the all requirements from **requirements.txt** by using pip.
```bash
pip install -r requirements.txt
```

## Usage [Windows]
**Make Sure You Have Installed Python 3.5+ Version, To Check Your Python Version on Your Command Prompt Give The Command ```python --version```**

For CLI ( Command Line Interface ):
```bash
python imghide.py
```
For GUI ( Graphical User Interface:
```bash
python imghide-gui.py
```

## Usage [Linux]
**Make Sure You Have Installed Python 3.5+ Version, To Check Your Python Version on Your Command Prompt Give The Command ```python --version```**

For CLI ( Command Line Interface ):
```bash
python3 imghide.py
```
For GUI ( Graphical User Interface:
```bash
python3 imghide-gui.py
```

---

[<img src="https://img.shields.io/github/followers/Majnu007?label=GitHub&logo=github&style=for-the-badge">](https://github.com/MajnuBhai007) &nbsp; [<img src="https://img.shields.io/badge/Queries%20%3F-Telegram-0088cc?style=for-the-badge&logo=telegram">](https://t.me/MajnuBhai007) <br />

[<img src="https://img.shields.io/badge/Queries%20%3F-Email-00a4e4?style=for-the-badge&logo=protonmail">](mailto:majnu007@protonmail.com) &nbsp; [<img src="https://img.shields.io/badge/Visit-My%20Website-0dd3ff?style=for-the-badge">](https://Majnu007.github.io)

---

## Credits:
* CLI Original Version By [TechRaj156](https://www.youtube.com/c/TechRaj156?sub_confirmation=1)
* GUI Versionn By [@Majnu007](https://github.com/Majnu007) 
* Original Script Was Inspired from this [Medium post](https://medium.com/better-programming/image-steganography-using-python-2250896e48b9)
* Watch [explanation video](https://youtu.be/_KX8ORUA_98)

---

## Thanks For Visiting


Changelog.txt
Version 1.0 [ 12-Feb-2021 ]:
	Nothing Changed

Version 1.1 [ 14-Feb-2021 ]:
	+ Simple UI Changes
	- Reduced Dependencies


Version 1.2 [ 16-Feb-2021 ]:
	+ Simple UI Changes
	+ Text in Password Entry Field is Hidden with '*'
	+ Added A Functioning 'Info Bar' at Bottom Left
	+ Fixed Some Minor Bugs
	- Reduced Dependencies

Version 1.3 [ 22-Feb-2021 ]:
	+ Simple UI Changes

imghide-cli.py
from PIL import Image
import os.path
from os import path
import math
from Crypto.Cipher import AES
from Crypto.Hash import SHA256
from Crypto import Random
import base64
from colorama import init
from termcolor import cprint 
from pyfiglet import figlet_format
from rich import print
from rich.console import Console
from rich.table import Table
import os
import getpass
from rich.progress import track
import sys

DEBUG = False
console = Console()


def encrypt(key, source, encode=True):
    key = SHA256.new(key).digest()  # use SHA-256 over our key to get a proper-sized AES key
    IV = Random.new().read(AES.block_size)  # generate IV
    encryptor = AES.new(key, AES.MODE_CBC, IV)
    padding = AES.block_size - len(source) % AES.block_size  # calculate needed padding
    source += bytes([padding]) * padding  # Python 2.x: source += chr(padding) * padding
    data = IV + encryptor.encrypt(source)  # store the IV at the beginning and encrypt
    return base64.b64encode(data).decode() if encode else data

def decrypt(key, source, decode=True):
    if decode:
        source = base64.b64decode(source.encode())
    key = SHA256.new(key).digest()  # use SHA-256 over our key to get a proper-sized AES key
    IV = source[:AES.block_size]  # extract the IV from the beginning
    decryptor = AES.new(key, AES.MODE_CBC, IV)
    data = decryptor.decrypt(source[AES.block_size:])  # decrypt
    padding = data[-1]  # pick the padding value from the end; Python 2.x: ord(data[-1])
    if data[-padding:] != bytes([padding]) * padding:  # Python 2.x: chr(padding) * padding
        raise ValueError("Invalid padding...")
    return data[:-padding]  # remove the padding


def convertToRGB(img):
	try:
		rgba_image = img
		rgba_image.load()
		background = Image.new("RGB", rgba_image.size, (255, 255, 255))
		background.paste(rgba_image, mask = rgba_image.split()[3])
		print("[yellow]Converted image to RGB [/yellow]")
		return background
	except Exception as e:
		print("[red]Couldn't convert image to RGB [/red]- %s"%e)

def getPixelCount(img):
	width, height = Image.open(img).size
	return width*height



def encodeImage(image,message,filename):
	with console.status("[green]Encoding image..") as status:
		try:
			width, height = image.size
			pix = image.getdata()

			current_pixel = 0
			tmp=0
			# three_pixels = []
			x=0
			y=0
			for ch in message:
				binary_value = format(ord(ch), '08b')
				
				# For each character, get 3 pixels at a time
				p1 = pix[current_pixel]
				p2 = pix[current_pixel+1]
				p3 = pix[current_pixel+2]

				three_pixels = [val for val in p1+p2+p3]

				for i in range(0,8):
					current_bit = binary_value[i]

					# 0 - Even
					# 1 - Odd
					if current_bit == '0':
						if three_pixels[i]%2!=0:
							three_pixels[i]= three_pixels[i]-1 if three_pixels[i]==255 else three_pixels[i]+1
					elif current_bit == '1':
						if three_pixels[i]%2==0:
							three_pixels[i]= three_pixels[i]-1 if three_pixels[i]==255 else three_pixels[i]+1

				current_pixel+=3
				tmp+=1

				#Set 9th value
				if(tmp==len(message)):
					# Make as 1 (odd) - stop reading
					if three_pixels[-1]%2==0:
						three_pixels[-1]= three_pixels[-1]-1 if three_pixels[-1]==255 else three_pixels[-1]+1
				else:
					# Make as 0 (even) - continue reading
					if three_pixels[-1]%2!=0:
						three_pixels[-1]= three_pixels[-1]-1 if three_pixels[-1]==255 else three_pixels[-1]+1


				if DEBUG:
					print("Character: ",ch)
					print("Binary: ",binary_value)
					print("Three pixels before mod: ",three_pixels)
					print("Three pixels after mod: ",three_pixels)


				three_pixels = tuple(three_pixels)
				
				st=0
				end=3

				for i in range(0,3):
					if DEBUG:
						print("Putting pixel at ",(x,y)," to ",three_pixels[st:end])

					image.putpixel((x,y), three_pixels[st:end])
					st+=3
					end+=3

					if (x == width - 1):
						x = 0
						y += 1
					else:
						x += 1

			encoded_filename = filename.split('.')[0] + "-enc.png"
			image.save(encoded_filename)
			print("\n")
			print("[yellow]Original File: [u]%s[/u][/yellow]"%filename)
			print("[green]Image encoded and saved as [u][bold]%s[/green][/u][/bold]"%encoded_filename)

		except Exception as e:
			print("[red]An error occured - [/red]%s"%e)
			sys.exit(0)



def decodeImage(image):
	with console.status("[green]Decoding image..") as status:
		try:
			pix = image.getdata()
			current_pixel = 0
			decoded=""
			while True:
				# Get 3 pixels each time
				binary_value=""
				p1 = pix[current_pixel]
				p2 = pix[current_pixel+1]
				p3 = pix[current_pixel+2]
				three_pixels = [val for val in p1+p2+p3]

				for i in range(0,8):
					if three_pixels[i]%2==0:
						# add 0
						binary_value+="0"
					elif three_pixels[i]%2!=0:
						# add 1
						binary_value+="1"


				#Convert binary value to ascii and add to string
				binary_value.strip()
				ascii_value = int(binary_value,2)
				decoded+=chr(ascii_value)
				current_pixel+=3

				if DEBUG:
					print("Binary: ",binary_value)
					print("Ascii: ",ascii_value)
					print("Character: ",chr(ascii_value))

				if three_pixels[-1]%2!=0:
					# stop reading
					break

			# print("Decoded: %s"%decoded)
			return decoded
		except Exception as e:
			print("[red]An error occured - [/red]%s"%e)
			sys.exit()




def test(img):
	image = Image.open(img)
	pix = image.load()
	print(pix[0])
	print(type(pix))

def print_credits():
	table = Table(show_header=True)
	table.add_column("Author",style="yellow")
	table.add_column("Contact",style="yellow")
	table.add_row("Teja Swaroop", "tejaswaroop156@gmail.com ")
	console.print(table)


def insertHeaders(img):
	pass

def main():
	print("[cyan]Choose one: [/cyan]")
	op = int(input("1. Encode\n2. Decode\n>>"))

	if op==1:
		print("[cyan]Image path (with extension): [/cyan]")
		img = input(">>")
		if(not(path.exists(img))):
			raise Exception("Image not found!")

		
		print("[cyan]Message to be hidden: [/cyan]")
		message = input(">>")
		if(len(message)*3 > getPixelCount(img)):
			raise Exception("Given message is too long to be encoded in the image.")


		password=""
		while 1:
			print("[cyan]Password to encrypt (leave empty if you want no password): [/cyan]")
			password = getpass.getpass(">>")
			if password=="":
				break
			print("[cyan]Re-enter Password: [/cyan]")
			confirm_password = getpass.getpass(">>")
			if(password!=confirm_password):
				print("[red]Passwords don't match try again [/red]")
			else:
				break

		cipher=""
		if password!="":
			cipher = encrypt(key=password.encode(),source=message.encode())
		else:
			cipher = message

		if DEBUG:
			print("[yellow]Encrypted : [/yellow]",cipher)

		image = Image.open(img)
		print("[yellow]Image Mode: [/yellow]%s"%image.mode)
		if image.mode!='RGB':
			image = convertToRGB(image)
		newimg = image.copy()
		encodeImage(image=newimg,message=cipher,filename=image.filename)

	elif op==2:
		print("[cyan]Image path (with extension): [/cyan]")
		img = input(">>")
		if(not(path.exists(img))):
			raise Exception("Image not found!")

		print("[cyan]Enter password (leave empty if no password): [/cyan]")
		password = getpass.getpass(">>")

		image = Image.open(img)

		cipher = decodeImage(image)
		# print(cipher)

		print()

		if DEBUG:
			print("[yellow]Cipher: %s[/yellow]"%cipher)

		decrypted=""

		if password!="":
			decrypted = decrypt(key=password.encode(),source=cipher)
		else:
			decrypted=cipher



		print("[green]Decoded Text: \n[bold]%s[/bold][/green]"%decrypted)






if __name__ == "__main__":
	os.system('cls' if os.name == 'nt' else 'clear')
	cprint(figlet_format('IMGHIDE!', font='starwars'),'yellow', attrs=['bold'])
	print_credits()
	print()
	print("[bold]IMGHIDE[/bold] allows you to hide texts inside an image. You can also protect these texts with a password using AES-256.")
	print()
	
	main()

imghide-gui.py
import tkinter as tk
from tkinter import messagebox
import PIL.Image
import PIL.ImageTk
from PIL import Image
from os import path
import math
from Crypto.Cipher import AES
from Crypto.Hash import SHA256
from Crypto import Random
import base64
from sys import exit
from random import choice

window = tk.Tk()
window.title('IMGHide GUI v1.3')
window.geometry('630x370')
window.configure(bg='deepskyblue4')
window.minsize(550, 370)
window.maxsize(630, 370)

im1 = PIL.Image.open(choice(["assets/header.png", "assets/header2.png"]))
header = PIL.ImageTk.PhotoImage(im1)

im2 = PIL.Image.open("assets/enc_button.png")
enc_button = PIL.ImageTk.PhotoImage(im2)

im3 = PIL.Image.open("assets/dec_button.png")
dec_button = PIL.ImageTk.PhotoImage(im3)


def encrypt(key, source, encode=True):
    key = SHA256.new(key).digest()  # use SHA-256 over our key to get a proper-sized AES key
    IV = Random.new().read(AES.block_size)  # generate IV
    encryptor = AES.new(key, AES.MODE_CBC, IV)
    padding = AES.block_size - len(source) % AES.block_size  # calculate needed padding
    source += bytes([padding]) * padding  # Python 2.x: source += chr(padding) * padding
    data = IV + encryptor.encrypt(source)  # store the IV at the beginning and encrypt
    return base64.b64encode(data).decode() if encode else data

def decrypt(key, source, decode=True):
    if decode:
        source = base64.b64decode(source.encode())
    key = SHA256.new(key).digest()  # use SHA-256 over our key to get a proper-sized AES key
    IV = source[:AES.block_size]  # extract the IV from the beginning
    decryptor = AES.new(key, AES.MODE_CBC, IV)
    data = decryptor.decrypt(source[AES.block_size:])  # decrypt
    padding = data[-1]  # pick the padding value from the end; Python 2.x: ord(data[-1])
    if data[-padding:] != bytes([padding]) * padding:  # Python 2.x: chr(padding) * padding
        messagebox.showerror("Error", "Invalid Padding Detected When Decrypting The Message !!!")
    return data[:-padding]  # remove the padding

def convertToRGB(img):
    try:
        rgba_image = img
        rgba_image.load()
        background = Image.new("RGB", rgba_image.size, (255, 255, 255))
        background.paste(rgba_image, mask = rgba_image.split()[3])
        info_label.config(text='$ Converted image to RGB')
        window.update()
        return background
    except Exception as e:
        info_label.config(text="$ Couldn't convert image to RGB")
        window.update()
        messagebox.showerror("Error", f"Couldn't convert image to RGB\n{e}")
        exit(1)

def getPixelCount(img):
    width, height = Image.open(img).size
    return width*height

def encodeImage(image,message,filename):
    info_label.config(text="$ Encoding The Image")
    window.update()
    try:
        width, height = image.size
        pix = image.getdata()

        current_pixel = 0
        tmp=0
        # three_pixels = []
        x=0
        y=0

        info_label.config(text="$ Encoding The Image.")
        window.update()

        for ch in message:
            info_label.config(text="$ Encoding The Image..")
            window.update()

            binary_value = format(ord(ch), '08b')
            # For each character, get 3 pixels at a time
            p1 = pix[current_pixel]
            p2 = pix[current_pixel+1]
            p3 = pix[current_pixel+2]

            three_pixels = [val for val in p1+p2+p3]
            for i in range(0,8):
                current_bit = binary_value[i]
            
                if current_bit == '0':
                    if three_pixels[i]%2!=0:
                        three_pixels[i]= three_pixels[i]-1 if three_pixels[i]==255 else three_pixels[i]+1
                elif current_bit == '1':
                    if three_pixels[i]%2==0:
                        three_pixels[i]= three_pixels[i]-1 if three_pixels[i]==255 else three_pixels[i]+1

            current_pixel+=3
            tmp+=1

            #Set 9th value
            if(tmp==len(message)):
                # Make as 1 (odd) - stop reading
                if three_pixels[-1]%2==0:
                    three_pixels[-1]= three_pixels[-1]-1 if three_pixels[-1]==255 else three_pixels[-1]+1
            else:
                # Make as 0 (even) - continue reading
                if three_pixels[-1]%2!=0:
                    three_pixels[-1]= three_pixels[-1]-1 if three_pixels[-1]==255 else three_pixels[-1]+1

            three_pixels = tuple(three_pixels)
                
            st=0
            end=3
            for i in range(0,3):
                image.putpixel((x,y), three_pixels[st:end])
                st+=3
                end+=3

                if (x == width - 1):
                    x = 0
                    y += 1
                else:
                    x += 1

            info_label.config(text="$ Encoding The Image...")
            window.update()

        encoded_filename = filename.split('.')[0] + "-encrypted.png"
        image.save(encoded_filename)
        info_label.config(text="$ Image Saved Successfully")
        window.update()
        messagebox.showinfo("Success", f"Image encoded and saved as {encoded_filename}\nOriginal filename {filename}")
    except Exception as e:
        messagebox.showerror("Error", f"An error occured\n{e}")
        exit(1)

def decodeImage(image):
    info_label.config(text="$ Decoding The Image")
    window.update()
    try:
        pix = image.getdata()
        current_pixel = 0
        decoded=""

        info_label.config(text="$ Decoding The Image.")
        window.update()
        while True:
            info_label.config(text="$ Decoding The Image..")
            window.update()
            # Get 3 pixels each time
            binary_value=""
            p1 = pix[current_pixel]
            p2 = pix[current_pixel+1]
            p3 = pix[current_pixel+2]
            three_pixels = [val for val in p1+p2+p3]

            for i in range(0,8):
                if three_pixels[i]%2==0:
                    binary_value+="0"
                elif three_pixels[i]%2!=0:
                    binary_value+="1"

            info_label.config(text="$ Decoding The Image...")
            window.update()

            #Convert binary value to ascii and add to string
            binary_value.strip()
            ascii_value = int(binary_value,2)
            decoded+=chr(ascii_value)
            current_pixel+=3
            info_label.config(text="$ Decoding The Image.")
            window.update()

            if three_pixels[-1]%2!=0:
                # stop reading
                break

        info_label.config(text="$ Image Decoded")
        window.update()
        return decoded
    except Exception as e:
        messagebox.showerror("Error", f"An error occured\n{e}")
        exit(1)

def insertHeaders(img):
    pass


def init_encode():
    c2.config(state='disabled')
    encode_button.config(state='disabled')

    img = path_var.get()
    if(not(path.exists(img))):
        messagebox.showerror("Error", "Image not found!\nGiven Image Name/Path is Invalid")
        exit(1)

    message = str(msg_var.get())
    if(len(message)*3 > getPixelCount(img)):
        messagebox.showerror("Error", "Given message is too long to be encoded in the image.\nPlease try another image with more pixels")
        exit(1)

    password = password_var.get()

    cipher=""
    if password!="":
        cipher = encrypt(key=password.encode(),source=message.encode())
    else:
        cipher = message

    image = Image.open(img)
    info_label.config(text=f"Image Mode: {image.mode}")
    window.update()
    if image.mode!='RGB':
        image = convertToRGB(image)
    newimg = image.copy()
    encodeImage(image=newimg,message=cipher,filename=image.filename)
    encode_button.config(state='normal')
    c2.config(state='normal')
    checkbox_var1.set(0)
    checkbox_var2.set(0)
    disable_checkbox()

def copytext(msg):
    r = tk.Tk()
    r.withdraw()
    r.clipboard_clear()
    r.clipboard_append(msg)
    r.update()

def init_decode():
    c1.config(state='disabled')
    encode_button.config(state='disabled')

    img = path_var.get()
    if(not(path.exists(img))):
        messagebox.showerror("Error", "Image not found!\nFilename/Path Provided is Invalid!")
        exit(1)

    password = str(password_var.get())
    image = Image.open(img)
    cipher = decodeImage(image)
    decrypted=""

    if password!="":
        decrypted = decrypt(key=password.encode(), source=cipher)
    else:
        decrypted=cipher

    response = messagebox.askyesno("Decoded Message", f'"{decrypted.decode("UTF-8")}"\n\nDo You Want To Copy The Message ?')
    if response == True:
        copytext(decrypted)

    c1.config(state='normal')
    encode_button.config(state='normal')
    checkbox_var1.set(0)
    checkbox_var2.set(0)
    disable_checkbox()

def disable_checkbox():
    global path_entry, msg_entry, password_entry, encode_button, label1, label2, label3, label4

    if (checkbox_var1.get() == 1) & (checkbox_var2.get() == 0):
        info_label.config(text='$ Encode Selected')
        c2.config(state='disabled')

        label1 = tk.Label(window,bg='deepskyblue4',fg='white', text='File Name/Path', font=('calibre',9,'normal'))
        label1.place(relx = 0.3, rely = 0.43, anchor = 'center')
        path_entry = tk.Entry(window,textvariable = path_var, font=('calibre',10,'normal'))
        path_entry.place(relx = 0.4, rely = 0.43, anchor = 'w')

        label2 = tk.Label(window,bg='deepskyblue4',fg='white', text=' Message', font=('calibre',9,'normal'))
        label2.place(relx = 0.33, rely = 0.53, anchor = 'center')
        msg_entry = tk.Entry(window,textvariable = msg_var, font=('calibre',10,'normal'))
        msg_entry.place(relx = 0.4, rely = 0.53, anchor = 'w')

        label3 = tk.Label(window,bg='deepskyblue4',fg='white', text='Password', font=('calibre',9,'normal'))
        label3.place(relx = 0.33, rely = 0.63, anchor = 'center')
        password_entry = tk.Entry(window,textvariable = password_var, show='*', font=('calibre',10,'normal'))
        password_entry.place(relx = 0.4, rely = 0.63, anchor = 'w')
        label4 = tk.Label(window,bg='deepskyblue4',fg='white',text='(Leave Empty For No Password)', font=('calibre',9,'normal'))
        label4.place(relx = 0.5, rely = 0.7, anchor = 'center')

        encode_button = tk.Button(window,image=enc_button,borderwidth = 2,relief="flat",command=init_encode)
        encode_button.place(relx = 0.4, rely = 0.825, anchor = 'w')

    elif (checkbox_var1.get() == 0) & (checkbox_var2.get() == 1):
        info_label.config(text='$ Decode Selected')
        c1.config(state='disabled')

        label1 = tk.Label(window,bg='deepskyblue4',fg='white', text='File Name/Path', font=('calibre',9,'normal'))
        label1.place(relx = 0.3, rely = 0.45, anchor = 'center')
        path_entry = tk.Entry(window,textvariable = path_var, font=('calibre',10,'normal'))
        path_entry.place(relx = 0.4, rely = 0.45, anchor = 'w')

        label2 = tk.Label(window,bg='deepskyblue4',fg='white', text='Password', font=('calibre',9,'normal'))
        label2.place(relx = 0.33, rely = 0.55, anchor = 'center')

        label3 = tk.Label(window, text='', font=('calibre',9,'normal'))
        password_entry = tk.Entry(window,textvariable=password_var, show='*', font=('calibre',10,'normal'))
        password_entry.place(relx = 0.4, rely = 0.55, anchor = 'w')

        label4 = tk.Label(window,bg='deepskyblue4',fg='white', text='(Leave Empty For No Password)', font=('calibre',9,'normal'))
        label4.place(relx = 0.5, rely = 0.65, anchor = 'center')

        msg_entry = tk.Entry(window,textvariable = msg_var, font=('calibre',10,'normal'))

        encode_button = tk.Button(window,image=dec_button,command=init_decode,borderwidth = 2,relief="flat")
        encode_button.place(relx = 0.4, rely = 0.8, anchor = 'w')

    elif (checkbox_var1.get() == 0) & (checkbox_var2.get() == 0):
        info_label.config(text='$ Nothing Selected')
        c1.config(state='normal')
        c2.config(state='normal')
        path_entry.delete('0',tk.END)
        path_entry.destroy()
        msg_entry.delete('0',tk.END)
        msg_entry.destroy()
        password_entry.delete('0',tk.END)
        password_entry.destroy()
        encode_button.destroy()
        label1.destroy()
        label2.destroy()
        label3.destroy()
        label4.destroy()
    else:
        info_label.config(text='$ Selected Nothing')

checkbox_var1 = tk.IntVar()
checkbox_var2 = tk.IntVar()
path_var = tk.StringVar()
password_var = tk.StringVar()
msg_var = tk.StringVar()

title = tk.Label(window, image=header, bg='deepskyblue4')
title.place(relx = 0.5, rely = 0.12, anchor = 'center')

c1 = tk.Checkbutton(window, text='Encode', bg='skyblue', variable=checkbox_var1, onvalue=1, offvalue=0, command=disable_checkbox)
c1.place(relx = 0.5, rely = 0.28, anchor = 'center')

c2 = tk.Checkbutton(window,text='Decode', bg='lawngreen', variable=checkbox_var2, onvalue=1, offvalue=0, command=disable_checkbox)
c2.place(relx = 0.5, rely = 0.35, anchor = 'center')

info_label = tk.Label(window, bg='black', fg='lawngreen',width=30, text='$ Everything Initialised', font=('calibre',9,'normal'))
info_label.place(relx = 0.0, rely = 1.0, anchor ='sw')

author_label = tk.Label(window, bg='black', fg='deep sky blue', width=30, text='GUI by @Majnu007 ( GitHub )   \nIMGHide by @TechRaj156 ( YouTube )', font=('TkHeadingFont',9,'normal'))
author_label.place(relx = 1, rely = 1.0, anchor ='se')

window.mainloop()